test/rql_test/connections/twisted_connection.py

```diff
@@ -755,18 +743,22 @@ def test_rate_limit(self):
         cursor_initial_size = len(cursor.items)
 
         # Wait for the second (pre-fetched) batch to arrive
-        yield cursor.items.wait_for_new_item() # Wait for the queue.
-        yield sleep(0.5) # Let's wait a little bit (to let the items aggregate them).
```

Comment:
Tests that rely on a sleep are fragile - got this working by using the cursor's underlying `waiting` list.